File: Go/Go_1000-/1374.go
// Go
// leetcode 1374
// https://leetcode.cn/problems/generate-a-string-with-characters-that-have-odd-counts/

func generateTheString(n int) string {
	tmp := make([]byte, n)
	for i := 1; i < n; i++ {
		tmp[i] = 'a'
	}
	if n&1 == 1 {
		tmp[0] = 'a'
	} else {
		tmp[0] = 'b'
	}

	return string(tmp)
}

File: cpp/cpp_1000-/1374.cc
// C++
// leetcode 1374
// https://leetcode.cn/problems/generate-a-string-with-characters-that-have-odd-counts/

class Solution {
public:
    string generateTheString(int n) {
    	if ((n & 1) == 1) {
    		// odd
    		return string(n, 'a');
    	}
    	return "a" + string(n-1, 'b');
    }
};